### DISCUSSION TITLE
Mobile view needs a full screen option.

### DISCUSSION THREAD

**Discussion by dopeytree** [blakeblackshear/frigate#13195] [Ideas]
Mobile view needs a full screen option.

Noticed desktop has some slightly unusual full screen mode where it's probably a java app or similar as has the timeline on the right and other webcams below.

Is there a way to just have the video feed full screen on mobile when a button is pressed?

-----------------------------------------------------------

Landscape view on an iPhone 15 Pro Max (6.69 inches)
![IMG_7344](https://github.com/user-attachments/assets/ba0b6918-e58a-4732-809c-93cafa78b298)

Portrait view on an iPhone 15 Pro Max (6.69 inches)
![IMG_7343](https://github.com/user-attachments/assets/43c4affd-d394-4ff7-80d9-2284e5043bf6)

-----------------------------------------------------------

Desktop View
<img width="1512" alt="Screenshot 2024-08-19 at 20 47 34" src="https://github.com/user-attachments/assets/2c26cb5c-c280-459c-a3ab-0748b4937863">

Fullscreen Desktop View
<img width="1512" alt="Screenshot 2024-08-19 at 20 47 38" src="https://github.com/user-attachments/assets/46bb5464-b80c-4071-bfcd-05705026d18a">

**Comment by NickM-27** [comment]
mobile does have a full screen view, however on iOS this is not supported due to iOS Safari not supporting the feature. Once it is supported by safari the button will show up

**Reply by dopeytree** [reply]
Hi yes I had guessed that as new frigate is using some strange full screen app mode.

My view would be how many of our frigate+ customers are using IOS etc.

Can we have a fall back mode that uses the basics of web design to detect the client & os.

Does the client actually want 1/3 of the mobile device to be a timeline?

Its easy enough to show the video stream from Go2RTC...

**Reply by NickM-27** [reply]
It's not strange, literally every single browser besides iOS supports it, even iPadOS supports it.

https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenElement#browser_compatibility

On iOS 18 there is a setting to enable it but it is disabled by default, so it shows Apple may add it in the future and at least now users will have an option.

We already use that to detect when to show the full screen button hence the reason the button isn't there.

**Reply by jake9190** [reply]
For anyone who wants to try this in iOS 18, go into settings -> Safari -> Advanced -> Feature Flags and enable Fullscreen API at the bottom.

This works in Safari, but not when in the progressive web app (at least for me).

**Comment by dopeytree** [comment]
also the OP raised a question which wasn't addressed - 'Is there a way to _just have the video feed full screen_ on mobile when a button is pressed?'

**Reply by NickM-27** [reply]
No, because without the timeline you couldn't move around in the video

**Reply by dopeytree** [reply]
You'd use the timeline in non full-screen mode?

Tap between the two. Like how browsing photos on a phone works full screen or tap out to view thumbnails.

The main objective is to view the video larger with the small screen estate available?

Normally we're only going full screen to review a clip that's been detected so the timeline is not such a priority?

Maybe in the future tap to hide timeline and have the video full screen.

**Reply by NickM-27** [reply]
> Normally we're only going full screen to review a clip that's been detected so the timeline is not needed?

A lot of feedback we received during the beta cycle says that is not true, many users wanted fullscreen so that the camera selector and other UI elements were not visible (making the video player and timeline larger) but the timeline was still very useful for navigating the recordings

**Reply by dopeytree** [reply]
Cool ok well always room to update in future. Anyway hopefully safari gets updated soon and apple don't do a silly one.

**Comment by broyuken** [comment]
Is there a way to get full screen working in the iOS HA app? I use the frigate proxy addon. Currently there is no full screen button, which I did enable in safari settings, and if I rotate my phone to landscape, it reloads the page back to the main frigate page.

**Reply by hawkeye217** [reply]
The fullscreen API is still not supported on Safari/iOS on iPhone: https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenElement#browser_compatibility

Installing Frigate as a PWA will give you more screen real estate: https://docs.frigate.video/configuration/pwa/

**Reply by NickM-27** [reply]
To be clear it is supported, but you have to enable it in Safari feature flags

**Reply by hawkeye217** [reply]
Ah yes. Settings --> Apps --> Safari --> Advanced --> Feature Flags --> Fullscreen API.

**Reply by broyuken** [reply]
I did that in safari, and have the full screen button there. But the HA app does not 

edit: and I’m not using the PWA because I need the frigate proxy to access frigate outside my network.

**Reply by dopeytree** [reply]
I have used an iframe on safari as seemed best user experience. 

The official frigate add on for homeassisant has odd navigation.

Using html code on the actual frigate service you can then have certain filters reapplied like cat or person etc. 

If you had a blank page except the iframe it should load in full screen when rotated?

Also recommend Tailscale full easy secure outside access to frigate & HA etc

<img width="1118" height="565" alt="Screenshot 2025-10-03 at 12 22 35" src="https://github.com/user-attachments/assets/a94d320c-c06d-42a3-a4dc-3e3e5a7938ad" />
